### PR TITLE
IGNITE-14149

### PR DESCRIPTION
**Pull request by ascherbakoff**

### CODE REVIEW

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/PeerId.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ */
+
+package org.apache.ignite.raft.client;
+
+import org.apache.ignite.network.NetworkMember;
+
+/**
+ * A participant of a replication group.
+ */
+public final class PeerId {
```
Comment (AMashenkov) [inline modules/raft-client/src/main/java/org/apache/ignite/raft/client/PeerId.java:25]: Name "Peer" looks more suitable to me.

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/message/impl/RemovePeersRequestImpl.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override public Builder setGroupId(String groupId) {
+        this.groupId = groupId;
+
+        return this;
+    }
+
+    @Override public Builder addPeer(PeerId peerId) {
+        if (!this.peersList.contains(peerId))
```
Comment (AMashenkov) [inline modules/raft-client/src/main/java/org/apache/ignite/raft/client/message/impl/RemovePeersRequestImpl.java:45]: Is this check really needed? May be Set be more suitable?
Also, a similar code in RemoveLearnersRequestImpl doesn't have one.

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/RaftGroupService.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    /**
+     * @return Current leader id or {@code null} if it has not been yet initialized.
+     */
+    @Nullable PeerId leader();
+
+    /**
+     * @return List of voting peers or {@code null} if it has not been yet initialized.
+     */
+    @Nullable List<PeerId> peers();
```
Comment (AMashenkov) [inline modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/RaftGroupService.java:73]: You use List as a result type. Does the order matter?
Guess the first list member is a leader.
If so, it worth to be mentioned in javadoc.

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/impl/RaftGroupServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+        LOG.log(System.Logger.Level.WARNING, msg, params);
+    }
+
+    private void error(String msg, Object... params) {
+        LOG.log(System.Logger.Level.ERROR, msg, params);
+    }
+
+    private void error(String msg, Exception e) {
+        LOG.log(System.Logger.Level.ERROR, msg, e);
+    }
```
Comment (AMashenkov) [inline modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/impl/RaftGroupServiceImpl.java:230]: Let's wrap System.Logger and move this to a separate class to keep code clean.
Later, the logger wrapper can be easily refactored.

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/PeerId.java (hunk 1 above)
Comment (ascherbakoff) [reply modules/raft-client/src/main/java/org/apache/ignite/raft/client/PeerId.java]: Ok

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/RaftGroupService.java (hunk 3 above)
Comment (ascherbakoff) [reply modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/RaftGroupService.java]: The position in the list reflects the time a peer has been added to the group. 
I prefer to have a List here for efficiency as well.

File: modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/impl/RaftGroupServiceImpl.java (hunk 4 above)
Comment (ascherbakoff) [reply modules/raft-client/src/main/java/org/apache/ignite/raft/client/service/impl/RaftGroupServiceImpl.java]: I've introduced LogWrapper and moved it to the core. It delegates to system logger.

File: pom.xml [hunk 5]
```diff
@@ -45,5 +45,6 @@
         <module>modules/rest</module>
         <module>modules/runner</module>
         <module>modules/schema</module>
+        <module>modules/raft-client</module>
```
Comment (AMashenkov) [inline pom.xml:48]: Let's keep alphabetical order here.

**Review by AMashenkov — APPROVED** [review]

**Comment by ascherbakoff** [comment]
Merged to master #cb518dea2d7b6c84aa8ee9c5c9cc6720db4a75c9

**Pull request closed without merging**